core/lib/src/rocket.rs

```diff
@@ -43,13 +55,21 @@ pub struct Rocket {
     shutdown_receiver: Option<mpsc::Receiver<()>>,
 }
 
+/// A readonly view into the current state of an unlaunched [`Rocket`].
+///
+/// The `Rocket` type only provides a single method for inspection,
+/// [`Rocket::inspect`], which returns this type. An `Inspector` instance can be
+/// used to access configuration, managed state, and mounted routes.
+#[derive(Copy, Clone)]
+pub struct Inspector<'r>(pub(crate) &'r RocketInner);
```

Comment:
What do we get out of wrapping `RocketInner` in the `Inspector` struct? Do we lose any safety by passing around `&Manifest` to end-users?

Reply:
Good point - privacy should be enough to make this safe. I didn't want to expose the name `RocketInner` or use the name `Inspector` as the underlying type name, but the name `Manifest` solves both of those problems.